core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerClientWithFileAccess.scala

```diff
@@ -0,0 +1,188 @@
+     *
+     * Does not verify that the returned file actually exists.
+     *
+     * @param containerId Id of the desired Docker container
+     * @return canonical location of the container's configuration file
+     */
+    protected def containerConfigFile(containerId: ContainerId) = {
+        new File(containerDirectory(containerId), "config.v2.json").getCanonicalFile()
+    }
+
+    /**
+     * Provides the log file of the specified Docker container written by
+     * Docker's JSON log driver.
+     *
+     * Assumes that the file has the well-known location and name.
+     *
+     * Does not verify that the returned file actually exists.
+     *
+     * @param containerId Id of the desired Docker container
+     * @return canonical location of the container's log file
+     */
+    protected def containerLogFile(containerId: ContainerId) =
+        new File(containerDirectory(containerId), s"${containerId.asString}-json.log").getCanonicalFile()
+
+    /**
+     * Provides the contents of the specified Docker container's configuration
+     * file as JSON object.
+     *
+     * @param configFile the container's configuration file in JSON format
+     * @return contents of configuration file as JSON object
+     */
+    protected def configFileContents(configFile: File): Future[JsObject] = Future {
+        blocking { // Needed due to synchronous file operations
+            val source = scala.io.Source.fromFile(configFile)
+            val config = try source.mkString finally source.close()
+            config.parseJson.asJsObject
+        }
+    }
+
+    /**
+     * Extracts the IP of the container from the local config file of the docker daemon.
+     *
+     * A container may have more than one network. The container has an
+     * IP address in each of these networks such that the network name
+     * is needed.
+     *
+     * @param id the id of the container to get the IP address from
+     * @param network name of the network to get the IP address from
+     * @return the ip address of the container
+     */
+    protected def ipAddressFromFile(id: ContainerId, network: String): Future[ContainerIp] = {
+        configFileContents(containerConfigFile(id)).map { json =>
+            val networks = json.fields("NetworkSettings").asJsObject.fields("Networks").asJsObject
+            val specifiedNetwork = networks.fields(network).asJsObject
+            val ipAddr = specifiedNetwork.fields("IPAddress")
+            ContainerIp(ipAddr.convertTo[String])
+        }
+    }
+
+    // See implemented trait for description
+    override def inspectIPAddress(id: ContainerId, network: String)(implicit transid: TransactionId): Future[ContainerIp] = {
+        ipAddressFromFile(id, network).recoverWith {
+            case _ => super.inspectIPAddress(id, network)
+        }
+    }
+
+    // See implemented trait for description
+    def rawContainerLogs(containerId: ContainerId, fromPos: Long): Future[ByteBuffer] = Future {
```

Comment:
The reasoning is, that `inspectIPAddress` truly is an override, while `rawContainerLogs` implements the trait only. We don't use `override` for trait implementations, if they don't override anything.